File: problems/1669_merge_in_between_linked_lists/solution.py
from shared.python.data_structures import ListNode


class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        prev = list1
        for i in range(a - 1):
            prev = prev.next
        end = prev
        for i in range(b - a + 2):
            end = end.next
        prev.next = list2
        tail = list2
        while tail.next:
            tail = tail.next
        tail.next = end
        return list1

File: problems/1669_merge_in_between_linked_lists/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    ListNode* mergeInBetween(ListNode* list1, int a, int b, ListNode* list2) {
        ListNode* prev = list1;
        for (int i = 0; i < a - 1; i++) prev = prev->next;
        ListNode* end = prev;
        for (int i = 0; i < b - a + 2; i++) end = end->next;
        prev->next = list2;
        ListNode* tail = list2;
        while (tail->next) tail = tail->next;
        tail->next = end;
        return list1;
    }
};
